server/channels/jobs/s3_path_migration/s3_path_migration.go

```diff
@@ -52,6 +52,9 @@ func MakeWorker(jobServer *jobs.JobServer, store store.Store, fileBackend filest
 
 func (worker *S3PathMigrationWorker) Run() {
 	mlog.Debug("Worker started", mlog.String("worker", worker.name))
+	// We have to re-assign the stop channel again, because
+	// it might happen that the job was restarted due to a config change.
+	worker.stop = make(chan bool, 1)
```

Comment:
Wandering whether we should set this right after closing the channel?

Reply:
We could yeah. But this is the pattern we follow in https://github.com/mattermost/enterprise/blob/3ddee2048e4f8d985da57a3ae8f43b075419eef4/elasticsearch/indexing_job.go#L106-L108. Just being consistent.

Reply:
I'd love to see this get pulled up into a common worker implementation so the jobs can focus on the business logic they own. But to clarify, what makes the elasticsearch indexer worker different in that guards against multiple runs / stops with `worker.closed`? Should we be consistent and do that here too?

Reply:
Yeah there's a difference. In the common worker implementation, we don't close the channel, but send a value to it. `chan <- true` and then listen for it on the other side. BUT the difference in this worker is that we need to listen to this stop signal at multiple places! So if we send only once, we don't know which for loop will consume it. The best approach here is to close the channel which will trigger all for loops to simply close.

The elasticsearch worker had this requirement, and this worker does too.

Other workers don't. So we don't close the channel. And hence everything works fine :)